test/e2e/customconfigs/skipjobswithoutqueuename_test.go

```diff
@@ -302,7 +302,7 @@ var _ = ginkgo.Describe("ManageJobsWithoutQueueName", ginkgo.Ordered, func() {
 			ginkgo.By("verifying that the pods of the deployment are gated", func() {
 				gomega.Eventually(func(g gomega.Gomega) {
 					pods := &corev1.PodList{}
-					g.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Namespace),
+					g.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Name),
```

Comment:
Interesting, can you provide some examples for what is ns.Namespace and ns.Name in this context? 

How did it even work?

Reply:
Namespace doesn't have Namespace field. This is cluster object. It is working because we have another filter `client.MatchingLabels(testDeploy.Spec.Selector.MatchLabels)`.